dask/dataframe/groupby.py

```diff
@@ -1161,10 +1165,39 @@ def _aca_agg(
         token = self._token_prefix + token
         levels = _determine_levels(self.index)
 
+        first_index = self.index if not isinstance(self.index, list) else self.index[0]
+        embarrassingly_parallel = df.known_divisions and df._contains_index_name(
+            first_index if not is_index_like(first_index) else first_index.name
+        )
+
+        if embarrassingly_parallel:
+            if len(set(df.divisions[:-1])) == len(df.divisions[:-1]):
```

Comment:
```suggestion
            if len(set(df.divisions[:-1])) != len(df.divisions[:-1]):
```
did you mean this?

Reply:
Also could make the list of unique divisions here instead of in the warning statement to avoid a little repetition